test/unit/nic_test.rb

```diff
@@ -144,6 +144,20 @@ def teardown
       end
     end
 
+    test "bmc requires MAC address if managed" do
+      bmc = FactoryGirl.build(:nic_bmc, :managed => true)
+      bmc.mac = ''
+      refute bmc.valid?
+      assert_includes bmc.errors.keys, :mac
+    end
+
+    test "bmc does not require MAC address if unmanaged" do
+      bmc = FactoryGirl.build(:nic_bmc, :managed => false)
```

Comment:
You can make the whole test one line if you prefer:
`assert FactoryGirl.build(:nic_bmc, :managed => false, :mac => '').valid?`